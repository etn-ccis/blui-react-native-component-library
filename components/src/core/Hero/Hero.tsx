import React, { useCallback } from 'react';
import { View, TouchableOpacity, ViewStyle, TextStyle, StyleProp, StyleSheet, ViewProps } from 'react-native';
import { Text } from 'react-native-paper';
import { ChannelValue, ChannelValueProps as ChannelValuePropsType } from '../ChannelValue';
import { $DeepPartial } from '@callstack/react-theme-provider';
import { Icon } from '../Icon';
import { IconSource } from '../__types__';
import { useFontScale } from '../__contexts__/font-scale-context';
import { ExtendedTheme, useExtendedTheme } from '@brightlayer-ui/react-native-themes';

type HeroStyles = {
    root?: ViewStyle;
    iconWrapper?: ViewStyle;
    values?: ViewStyle;
    label?: TextStyle;
};

const makeStyles = (theme: ExtendedTheme, fontScale: number): StyleSheet.NamedStyles<HeroStyles> =>
    StyleSheet.create({
        root: {
            flex: 1,
            alignItems: 'center',
            justifyContent: 'center',
            paddingHorizontal: 8,
            paddingVertical: 16,
        },
        iconWrapper: {
            padding: 0,
            marginBottom: 8 * fontScale,
            alignItems: 'center',
            justifyContent: 'center',
            overflow: 'hidden',
            borderRadius: 50,
        },
        values: {
            flexDirection: 'row',
            justifyContent: 'center',
            alignItems: 'center',
            maxWidth: '100%',
        },
        label: {
            width: '100%',
            overflow: 'hidden',
            textAlign: 'center',
            fontFamily: 'OpenSans-SemiBold',
            fontWeight: '600',
            color: theme.colors.onSurfaceVariant,
        },
    });

export type HeroProps = ViewProps & {
    /** The text shown below the `ChannelValue` */
    label: string;

    /** The primary icon */
    icon?: IconSource;

    /** The size of the primary icon (min 10px)
     *
     * Default: 36
     */
    iconSize?: number;

    /** Color override for the row icon
     *
     * Default: theme.colors.onSurfaceVariant
     */
    iconColor?: string;

    /** The color used behind the primary icon
     *
     * Default: 'transparent
     */
    iconBackgroundColor?: string;

    /** Props to be passed through to ChannelValue child component */
    ChannelValueProps?: ChannelValuePropsType;

    /** Callback to execute when the drawer header is pressed */
    onPress?: () => void;

    /** Style overrides for internal elements. The styles you provide will be combined with the default styles. */
    styles?: {
        root?: StyleProp<ViewStyle>;
        iconWrapper?: StyleProp<ViewStyle>;
        values?: StyleProp<ViewStyle>;
        label?: StyleProp<TextStyle>;
    };

    /** Theme value overrides specific to this component. */
    theme?: $DeepPartial<ExtendedTheme>;
};

/**
 * [Hero](https://brightlayer-ui-components.github.io/react-native/?path=/info/components-documentation--hero) component
 *
 * The `<Hero>` component displays a particular icon, value/units, and a label. The icon property will accept any valid component - this will typically be a Material icon, [Brightlayer UI icon](https://github.com/etn-ccis/blui-icons), or [Progress Icon](https://github.com/etn-ccis/blui-progress-icons). It will also accept Text/Emoji values.
 */
export const Hero: React.FC<HeroProps> = (props) => {
    const {
        label,
        ChannelValueProps,
        onPress,
        icon,
        iconColor,
        iconSize,
        children,
        iconBackgroundColor,
        styles = {},
        style,
        theme: themeOverride,
        ...viewProps
    } = props;

    const theme = useExtendedTheme(themeOverride);
    const fontScale = useFontScale();
    const defaultStyles = makeStyles(theme, fontScale);

    const normalizeIconSize = useCallback((): number => {
        if (!iconSize) return 36;
        return Math.max(10, Math.min(48, iconSize));
    }, [iconSize]);

    const getColor = useCallback(
        (color: string | undefined): string => color || theme.colors.onSurfaceVariant,
        [theme]
    );

    const getIcon = useCallback((): JSX.Element | undefined => {
        if (icon) {
            return <Icon source={icon} size={normalizeIconSize()} color={getColor(iconColor)} />;
        }
    }, [icon, normalizeIconSize, getColor, iconColor]);

    return (
        <TouchableOpacity
            onPress={onPress}
            disabled={!onPress}
            style={[defaultStyles.root, styles.root, style]}
            {...viewProps}
        >
            <View
                style={[
                    defaultStyles.iconWrapper,
                    {
                        backgroundColor: iconBackgroundColor || 'rgba(255, 255, 255, 0)',
                        height: typeof iconSize === 'number' ? normalizeIconSize() : iconSize ?? 36 * fontScale,
                        width: typeof iconSize === 'number' ? normalizeIconSize() : iconSize ?? 36 * fontScale,
                    },
                    styles.iconWrapper,
                ]}
            >
                {getIcon()}
            </View>
            <View style={[defaultStyles.values, styles.values]}>
                {!children && !!ChannelValueProps?.value && <ChannelValue fontSize={22} {...ChannelValueProps} />}
                {children}
            </View>
            <Text
                variant={'titleMedium'}
                style={[defaultStyles.label, styles.label]}
                numberOfLines={1}
                ellipsizeMode={'tail'}
            >
                {label}
            </Text>
        </TouchableOpacity>
    );
};
