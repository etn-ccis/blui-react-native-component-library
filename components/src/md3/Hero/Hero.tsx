import React, { useCallback } from 'react';
import { View, TouchableOpacity, ViewStyle, TextStyle, StyleProp, StyleSheet, ViewProps } from 'react-native';
import { useTheme, MD3Theme, Text } from 'react-native-paper';
import { ChannelValue, ChannelValueProps as ChannelValuePropsType } from '../ChannelValue';
import { $DeepPartial } from '@callstack/react-theme-provider';
import { Icon } from '../Icon';
import { IconSource } from '../__types__';
import { useFontScale } from '../__contexts__/font-scale-context';

type HeroStyles = {
    root?: ViewStyle;
    iconWrapper?: ViewStyle;
    values?: ViewStyle;
    label?: TextStyle;
};

const makeStyles = (theme: MD3Theme, fontScale: number): StyleSheet.NamedStyles<HeroStyles> =>
    StyleSheet.create({
        root: {
            flex: 1,
            alignItems: 'center',
            justifyContent: 'center',
            paddingHorizontal: 8,
            paddingVertical: 16,
        },
        iconWrapper: {
            padding: 0,
            marginBottom: 8 * fontScale,
            alignItems: 'center',
            justifyContent: 'center',
            overflow: 'hidden',
            height: 36 * fontScale,
            width: 36 * fontScale,
        },
        values: {
            flexDirection: 'row',
            justifyContent: 'center',
            alignItems: 'center',
            maxWidth: '100%',
        },
        label: {
            width: '100%',
            overflow: 'hidden',
            textAlign: 'center',
            // fontFamily: 'OpenSans-SemiBold',
            // fontWeight: '600',
            color: theme.colors.onSurfaceVariant,
        },
    });

export type HeroProps = ViewProps & {
    label: string;
    icon?: IconSource;
    iconSize?: number;
    iconColor?: string;
    iconBackgroundColor?: string;
    ChannelValueProps?: ChannelValuePropsType;
    onPress?: () => void;
    styles?: {
        root?: StyleProp<ViewStyle>;
        iconWrapper?: StyleProp<ViewStyle>;
        values?: StyleProp<ViewStyle>;
        label?: StyleProp<TextStyle>;
    };
    theme?: $DeepPartial<MD3Theme>;
};

export const Hero: React.FC<HeroProps> = (props) => {
    const {
        label,
        ChannelValueProps,
        onPress,
        icon,
        iconColor,
        iconSize,
        children,
        styles = {},
        style,
        theme: themeOverride,
        ...viewProps
    } = props;

    const theme = useTheme(themeOverride);
    const fontScale = useFontScale();
    const defaultStyles = makeStyles(theme, fontScale);

    const normalizeIconSize = useCallback((): number => {
        if (!iconSize) return 36;
        return Math.max(10, Math.min(48, iconSize));
    }, [iconSize]);
    // @TODO update the color once the theme creation is complete
    const getColor = useCallback((color: string | undefined): string => color || theme.colors.onSurface, [theme]);

    const getIcon = useCallback((): JSX.Element | undefined => {
        if (icon) {
            return <Icon source={icon} size={normalizeIconSize()} color={getColor(iconColor)} />;
        }
    }, [icon, normalizeIconSize, getColor, iconColor]);

    return (
        <TouchableOpacity
            onPress={onPress}
            disabled={!onPress}
            style={[defaultStyles.root, styles.root, style]}
            {...viewProps}
        >
            <View style={[defaultStyles.iconWrapper, styles.iconWrapper]}>{getIcon()}</View>
            <View style={[defaultStyles.values, styles.values]}>
                {!children && !!ChannelValueProps?.value && <ChannelValue fontSize={20} {...ChannelValueProps} />}
                {children}
            </View>
            <Text
                variant={'titleMedium'}
                style={[defaultStyles.label, styles.label]}
                numberOfLines={1}
                ellipsizeMode={'tail'}
            >
                {label}
            </Text>
        </TouchableOpacity>
    );
};
