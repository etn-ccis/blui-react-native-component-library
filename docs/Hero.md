# Hero Components

Hero items are used to call attention to particular values.

<img width="500" alt="Hero Banner with heroes" src="./images/hero.png">

## Hero

The `<Hero>` component displays a particular icon / icon-like component, value/units, and a label.

The value section of the Hero utilizes a [`<ChannelValue>`](./ChannelValue.md) component. To display a single simple value, the information can be passed as `ChannelValueProps`. For more complex values (such as a duration that displays hours and minutes), you can pass in `<ChannelValue>` components as children and they will be displayed inline.

### Hero Usage

```tsx
import { Hero } from '@brightlayer-ui/react-native-components';

<Hero
    label={'Charge'}
    icon={{ family: 'brightlayer-ui', name: 'battery' }}
    ChannelValueProps={{ value: 100, units: '%' }}
/>;
```

### Hero API

<div style="overflow: auto">

| Prop Name           | Description                                | Type                                   | Required | Default                |
| ------------------- | ------------------------------------------ | -------------------------------------- | -------- | ---------------------- |
| ChannelValueProps   | Props passed through to ChannelValue child | `ChannelValueProps`                    | no       |                        |
| icon                | A component to render for the primary icon | [`IconSource`](./Icons.md)             | yes      |                        |
| iconBackgroundColor | The color behind the primary icon          | `string`                               | no       | transparent |
| iconColor           | The color of the primary icon              | `string`                               | no       | `theme.colors.text`    |
| iconSize            | The size of the primary icon (10-48)       | `number`                               | no       | 36                     |
| label               | The text shown below the `ChannelValue`    | `string`                               | yes      |                        |
| onPress             | A function to execute when pressed         | `function`                             | no       |                        |
| theme               | Theme value overrides                      | `$DeepPartial<ReactNativePaper.Theme>` | no       |                        |

</div>

### Styles

You can override the internal styles used by Brightlayer UI by passing a `styles` prop. It supports the following keys:

| Name        | Description                                  |
| ----------- | -------------------------------------------- |
| root        | Styles applied to the root element           |
| iconWrapper | Styles applied to the icon wrapper element   |
| label       | Styles applied to the label element          |
| values      | Styles applied to the row of `ChannelValue`s |

## Hero Banner

The `<HeroBanner>` component is a simple wrapper component that is used to contain `<Hero>`s. It creates the flex container and sets up the spacing rules to display them. It accepts up to four `<Hero>` components by default as children. Any children after the `limit` will not be rendered.

### Hero Banner Usage

```tsx
import { Hero, HeroBanner } from '@brightlayer-ui/react-native-components';

<HeroBanner divider>
    <Hero label={'Hero One'} icon={{ family: 'brightlayer-ui', name: 'battery' }} />
    <Hero label={'Hero Two'} icon={{ family: 'brightlayer-ui', name: 'battery' }} />
    <Hero label={'Hero Three'} icon={{ family: 'brightlayer-ui', name: 'battery' }} />
    <Hero label={'Hero Four'} icon={{ family: 'brightlayer-ui', name: 'battery' }} />
</HeroBanner>;
```

### Hero Banner API

<div style="overflow: auto">

| Prop Name | Description                                         | Type      | Required | Default |
| --------- | --------------------------------------------------- | --------- | -------- | ------- |
| divider   | Whether to show a dividing line below the subheader | `boolean` | no       | `false` |

</div>

### Styles

You can override the internal styles used by Brightlayer UI by passing a `styles` prop. It supports the following keys:

| Name    | Description                           |
| ------- | ------------------------------------- |
| root    | Styles applied to the root element    |
| divider | Styles applied to the divider element |
