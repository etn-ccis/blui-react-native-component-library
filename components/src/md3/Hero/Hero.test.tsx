import React from 'react';
import { cleanup, fireEvent, render } from '@testing-library/react-native';
import TestRenderer from 'react-test-renderer';
import { Hero } from './Hero';

describe('Hero', () => {
    afterEach(cleanup);

    it('renders correctly with label', () => {
        const tree = TestRenderer.create(<Hero label="Hero Label" />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders correctly with icon', () => {
        const tree = TestRenderer.create(<Hero label="Hero Label" icon="your-icon-source" />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders correctly with custom icon size and color', () => {
        const tree = TestRenderer.create(
            <Hero label="Hero Label" icon="your-icon-source" iconSize={30} iconColor="blue" />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders correctly with icon background color', () => {
        const tree = TestRenderer.create(
            <Hero label="Hero Label" icon="your-icon-source" iconBackgroundColor="yellow" />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders correctly with ChannelValueProps', () => {
        const tree = TestRenderer.create(<Hero label="Hero Label" ChannelValueProps={{ value: 100 }} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('calls onPress when pressed', () => {
        const onPressMock = jest.fn();
        const { getByText } = render(<Hero label="Hero Label" onPress={onPressMock} />);

        fireEvent.press(getByText('Hero Label'));
        expect(onPressMock).toHaveBeenCalled();
    });
    it('renders correctly with custom styles', () => {
        const customStyles = {
            root: { backgroundColor: 'red' },
            iconWrapper: { borderColor: 'green' },
            values: { padding: 10 },
            label: { color: 'blue' },
        };

        const tree = TestRenderer.create(<Hero label="Hero Label" styles={customStyles} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
