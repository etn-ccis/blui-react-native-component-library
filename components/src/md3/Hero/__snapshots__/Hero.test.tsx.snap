// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero renders correctly with ChannelValueProps 1`] = `
<View
  accessibilityState={
    {
      "disabled": true,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "opacity": 1,
      "paddingHorizontal": 8,
      "paddingVertical": 16,
    }
  }
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "height": 72,
          "justifyContent": "center",
          "marginBottom": 16,
          "overflow": "hidden",
          "padding": 0,
          "width": 72,
        },
        {
          "backgroundColor": "rgba(255, 251, 254, 1)",
        },
        undefined,
      ]
    }
  />
  <View
    style={
      [
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
          "maxWidth": "100%",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "flexDirection": "row",
            "maxWidth": "100%",
          },
          undefined,
          undefined,
        ]
      }
    >
      <Text
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          [
            {
              "textAlign": "left",
            },
            {
              "color": "rgba(28, 27, 31, 1)",
              "writingDirection": "ltr",
            },
            [
              {
                "fontFamily": "System",
                "fontSize": 16,
                "fontWeight": "400",
                "letterSpacing": 0.15,
                "lineHeight": 24,
              },
              [
                {
                  "color": "rgba(28, 27, 31, 1)",
                  "fontFamily": "OpenSans-Regular",
                  "fontSize": 22,
                  "fontWeight": "600",
                },
              ],
            ],
          ]
        }
        testID="text-wrapper"
      >
        <Text
          style={
            [
              {
                "textAlign": "left",
              },
              {
                "color": "rgba(28, 27, 31, 1)",
                "writingDirection": "ltr",
              },
              [
                {
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": "400",
                  "letterSpacing": 0.15,
                  "lineHeight": 24,
                },
                [
                  {
                    "color": "rgba(28, 27, 31, 1)",
                    "fontFamily": "OpenSans-Regular",
                    "fontSize": 22,
                    "fontWeight": "600",
                  },
                  undefined,
                ],
              ],
            ]
          }
        >
          100
        </Text>
      </Text>
    </View>
  </View>
  <Text
    ellipsizeMode="tail"
    numberOfLines={1}
    style={
      [
        {
          "textAlign": "left",
        },
        {
          "color": "rgba(28, 27, 31, 1)",
          "writingDirection": "ltr",
        },
        [
          {
            "fontFamily": "System",
            "fontSize": 16,
            "fontWeight": "500",
            "letterSpacing": 0.15,
            "lineHeight": 24,
          },
          [
            {
              "color": "rgba(73, 69, 79, 1)",
              "overflow": "hidden",
              "textAlign": "center",
              "width": "100%",
            },
            undefined,
          ],
        ],
      ]
    }
  >
    Hero Label
  </Text>
</View>
`;

exports[`Hero renders correctly with custom icon size and color 1`] = `
<View
  accessibilityState={
    {
      "disabled": true,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "opacity": 1,
      "paddingHorizontal": 8,
      "paddingVertical": 16,
    }
  }
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "height": 72,
          "justifyContent": "center",
          "marginBottom": 16,
          "overflow": "hidden",
          "padding": 0,
          "width": 72,
        },
        {
          "backgroundColor": "rgba(255, 251, 254, 1)",
        },
        undefined,
      ]
    }
  >
    <Text
      style={
        [
          {
            "textAlign": "left",
          },
          {
            "color": "rgba(28, 27, 31, 1)",
            "writingDirection": "ltr",
          },
          [
            {
              "fontFamily": "System",
              "fontSize": 14,
              "fontWeight": "400",
              "letterSpacing": 0.25,
              "lineHeight": 20,
            },
            [
              {
                "color": "blue",
                "fontSize": 30,
              },
            ],
          ],
        ]
      }
    >
      your-icon-source
    </Text>
  </View>
  <View
    style={
      [
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
          "maxWidth": "100%",
        },
        undefined,
      ]
    }
  />
  <Text
    ellipsizeMode="tail"
    numberOfLines={1}
    style={
      [
        {
          "textAlign": "left",
        },
        {
          "color": "rgba(28, 27, 31, 1)",
          "writingDirection": "ltr",
        },
        [
          {
            "fontFamily": "System",
            "fontSize": 16,
            "fontWeight": "500",
            "letterSpacing": 0.15,
            "lineHeight": 24,
          },
          [
            {
              "color": "rgba(73, 69, 79, 1)",
              "overflow": "hidden",
              "textAlign": "center",
              "width": "100%",
            },
            undefined,
          ],
        ],
      ]
    }
  >
    Hero Label
  </Text>
</View>
`;

exports[`Hero renders correctly with custom styles 1`] = `
<View
  accessibilityState={
    {
      "disabled": true,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "red",
      "flex": 1,
      "justifyContent": "center",
      "opacity": 1,
      "paddingHorizontal": 8,
      "paddingVertical": 16,
    }
  }
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "height": 72,
          "justifyContent": "center",
          "marginBottom": 16,
          "overflow": "hidden",
          "padding": 0,
          "width": 72,
        },
        {
          "backgroundColor": "rgba(255, 251, 254, 1)",
        },
        {
          "borderColor": "green",
        },
      ]
    }
  />
  <View
    style={
      [
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
          "maxWidth": "100%",
        },
        {
          "padding": 10,
        },
      ]
    }
  />
  <Text
    ellipsizeMode="tail"
    numberOfLines={1}
    style={
      [
        {
          "textAlign": "left",
        },
        {
          "color": "rgba(28, 27, 31, 1)",
          "writingDirection": "ltr",
        },
        [
          {
            "fontFamily": "System",
            "fontSize": 16,
            "fontWeight": "500",
            "letterSpacing": 0.15,
            "lineHeight": 24,
          },
          [
            {
              "color": "rgba(73, 69, 79, 1)",
              "overflow": "hidden",
              "textAlign": "center",
              "width": "100%",
            },
            {
              "color": "blue",
            },
          ],
        ],
      ]
    }
  >
    Hero Label
  </Text>
</View>
`;

exports[`Hero renders correctly with icon 1`] = `
<View
  accessibilityState={
    {
      "disabled": true,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "opacity": 1,
      "paddingHorizontal": 8,
      "paddingVertical": 16,
    }
  }
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "height": 72,
          "justifyContent": "center",
          "marginBottom": 16,
          "overflow": "hidden",
          "padding": 0,
          "width": 72,
        },
        {
          "backgroundColor": "rgba(255, 251, 254, 1)",
        },
        undefined,
      ]
    }
  >
    <Text
      style={
        [
          {
            "textAlign": "left",
          },
          {
            "color": "rgba(28, 27, 31, 1)",
            "writingDirection": "ltr",
          },
          [
            {
              "fontFamily": "System",
              "fontSize": 14,
              "fontWeight": "400",
              "letterSpacing": 0.25,
              "lineHeight": 20,
            },
            [
              {
                "color": "rgba(73, 69, 79, 1)",
                "fontSize": 36,
              },
            ],
          ],
        ]
      }
    >
      your-icon-source
    </Text>
  </View>
  <View
    style={
      [
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
          "maxWidth": "100%",
        },
        undefined,
      ]
    }
  />
  <Text
    ellipsizeMode="tail"
    numberOfLines={1}
    style={
      [
        {
          "textAlign": "left",
        },
        {
          "color": "rgba(28, 27, 31, 1)",
          "writingDirection": "ltr",
        },
        [
          {
            "fontFamily": "System",
            "fontSize": 16,
            "fontWeight": "500",
            "letterSpacing": 0.15,
            "lineHeight": 24,
          },
          [
            {
              "color": "rgba(73, 69, 79, 1)",
              "overflow": "hidden",
              "textAlign": "center",
              "width": "100%",
            },
            undefined,
          ],
        ],
      ]
    }
  >
    Hero Label
  </Text>
</View>
`;

exports[`Hero renders correctly with icon background color 1`] = `
<View
  accessibilityState={
    {
      "disabled": true,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "opacity": 1,
      "paddingHorizontal": 8,
      "paddingVertical": 16,
    }
  }
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "height": 72,
          "justifyContent": "center",
          "marginBottom": 16,
          "overflow": "hidden",
          "padding": 0,
          "width": 72,
        },
        {
          "backgroundColor": "yellow",
        },
        undefined,
      ]
    }
  >
    <Text
      style={
        [
          {
            "textAlign": "left",
          },
          {
            "color": "rgba(28, 27, 31, 1)",
            "writingDirection": "ltr",
          },
          [
            {
              "fontFamily": "System",
              "fontSize": 14,
              "fontWeight": "400",
              "letterSpacing": 0.25,
              "lineHeight": 20,
            },
            [
              {
                "color": "rgba(73, 69, 79, 1)",
                "fontSize": 36,
              },
            ],
          ],
        ]
      }
    >
      your-icon-source
    </Text>
  </View>
  <View
    style={
      [
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
          "maxWidth": "100%",
        },
        undefined,
      ]
    }
  />
  <Text
    ellipsizeMode="tail"
    numberOfLines={1}
    style={
      [
        {
          "textAlign": "left",
        },
        {
          "color": "rgba(28, 27, 31, 1)",
          "writingDirection": "ltr",
        },
        [
          {
            "fontFamily": "System",
            "fontSize": 16,
            "fontWeight": "500",
            "letterSpacing": 0.15,
            "lineHeight": 24,
          },
          [
            {
              "color": "rgba(73, 69, 79, 1)",
              "overflow": "hidden",
              "textAlign": "center",
              "width": "100%",
            },
            undefined,
          ],
        ],
      ]
    }
  >
    Hero Label
  </Text>
</View>
`;

exports[`Hero renders correctly with label 1`] = `
<View
  accessibilityState={
    {
      "disabled": true,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "opacity": 1,
      "paddingHorizontal": 8,
      "paddingVertical": 16,
    }
  }
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "height": 72,
          "justifyContent": "center",
          "marginBottom": 16,
          "overflow": "hidden",
          "padding": 0,
          "width": 72,
        },
        {
          "backgroundColor": "rgba(255, 251, 254, 1)",
        },
        undefined,
      ]
    }
  />
  <View
    style={
      [
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
          "maxWidth": "100%",
        },
        undefined,
      ]
    }
  />
  <Text
    ellipsizeMode="tail"
    numberOfLines={1}
    style={
      [
        {
          "textAlign": "left",
        },
        {
          "color": "rgba(28, 27, 31, 1)",
          "writingDirection": "ltr",
        },
        [
          {
            "fontFamily": "System",
            "fontSize": 16,
            "fontWeight": "500",
            "letterSpacing": 0.15,
            "lineHeight": 24,
          },
          [
            {
              "color": "rgba(73, 69, 79, 1)",
              "overflow": "hidden",
              "textAlign": "center",
              "width": "100%",
            },
            undefined,
          ],
        ],
      ]
    }
  >
    Hero Label
  </Text>
</View>
`;
